Python/Product/PythonTools/PythonToolsPackage.cs

```diff
@@ -383,70 +386,61 @@ internal sealed class PythonToolsPackage : CommonPackage, IVsComponentSelectorPr
-            services.AddService(typeof(IPythonToolsToolWindowService), this, promote: true);
-            services.AddService(typeof(PythonLanguageInfo), (container, serviceType) => new PythonLanguageInfo(container), true);
-            services.AddService(typeof(PythonToolsService), PythonToolsService.CreateService, promote: true);
-            services.AddService(typeof(ErrorTaskProvider), ErrorTaskProvider.CreateService, promote: true);
-            services.AddService(typeof(CommentTaskProvider), CommentTaskProvider.CreateService, promote: true);
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             var solutionEventListener = new SolutionEventsListener(this);
             solutionEventListener.StartListeningForChanges();
-
-            services.AddService(typeof(SolutionEventsListener), solutionEventListener, promote: true);
-
-            // Register custom debug event service
-            var customDebuggerEventHandler = new CustomDebuggerEventHandler(this);
-            services.AddService(customDebuggerEventHandler.GetType(), customDebuggerEventHandler, promote: true);
-
+            AddService(solutionEventListener, true);
+            
             // Enable the mixed-mode debugger UI context
             UIContext.FromUIContextGuid(DkmEngineId.NativeEng).IsActive = true;
 
             // Add our command handlers for menu (commands must exist in the .vsct file)
-            RegisterCommands(new Command[] { 
-                new OpenReplCommand(this, (int)PkgCmdIDList.cmdidReplWindow),
-                new OpenReplCommand(this, (int)PythonConstants.OpenInteractiveForEnvironment),
-                new OpenDebugReplCommand(this), 
-                new ExecuteInReplCommand(this), 
-                new SendToReplCommand(this), 
-                new FillParagraphCommand(this), 
-                new DiagnosticsCommand(this),
-                new RemoveImportsCommand(this, true),
-                new RemoveImportsCommand(this, false),
-                new OpenInterpreterListCommand(this),
-                new ImportWizardCommand(this),
-                new ImportCoverageCommand(this),
-                new ShowPythonViewCommand(this),
-                new ShowCppViewCommand(this),
-                new ShowNativePythonFrames(this),
-                new UsePythonStepping(this),
-                new AzureExplorerAttachDebuggerCommand(this),
-                new OpenWebUrlCommand(this, "https://go.microsoft.com/fwlink/?linkid=832525", PkgCmdIDList.cmdidWebPythonAtMicrosoft),
-                new OpenWebUrlCommand(this, Strings.IssueTrackerUrl, PkgCmdIDList.cmdidWebPTVSSupport),
-                new OpenWebUrlCommand(this, "https://go.microsoft.com/fwlink/?linkid=832517", PkgCmdIDList.cmdidWebDGProducts),
-            }, GuidList.guidPythonToolsCmdSet);
+            RegisterCommands(GuidList.guidPythonToolsCmdSet
+                , new OpenReplCommand(this, (int)PkgCmdIDList.cmdidReplWindow)
```

Comment:
What's with the comma at the start of the line? We don't use this style anywhere else, and I don't like it.

Reply:
Sorry, auto formatting on copy-paste